packages/pouchdb-core/src/adapter.js

```diff
@@ -620,23 +619,22 @@ AbstractPouchDB.prototype.get =
       var attachments = doc._attachments;
       var count = Object.keys(attachments).length;
       if (count === 0) {
-        return callback(null, doc);
+        return cb(null, doc);
       }
       Object.keys(attachments).forEach(function (key) {
         this._getAttachment(doc._id, key, attachments[key], {
           // Previously the revision handling was done in adapter.js
           // getAttachment, however since idb-next doesnt we need to
           // pass the rev through
           rev: doc._rev,
-          binary: opts.binary,
-          ctx: ctx
```

Comment:
If we can't do a full `get()` inside of a single transaction then that may be a bug; it could introduce race conditions if the transitions end up being interleaved. Based on your statement, I'm confused why the tests are even passing currently in Firefox if this is broken.

Ideally I think we would do this in a single transaction and then do `readBlobData()` after the transaction has ended.